router/internal/circuit/manager.go

```diff
@@ -0,0 +1,108 @@
+package circuit
+
+import (
+	"fmt"
+	"github.com/cep21/circuit/v4"
+	"github.com/cep21/circuit/v4/closers/hystrix"
+	nodev1 "github.com/wundergraph/cosmo/router/gen/proto/wg/cosmo/node/v1"
+	"github.com/wundergraph/cosmo/router/pkg/config"
+	"github.com/wundergraph/cosmo/router/pkg/metric"
+	"go.opentelemetry.io/otel/attribute"
+)
+
+type Manager struct {
+	// We maintain separate circuit breakers for each subgraph
+	circuits map[string]*circuit.Circuit
+}
+
+func (c *Manager) GetCircuitBreaker(name string) *circuit.Circuit {
+	if c == nil {
+		return nil
+	}
+
+	// Since we should not ever write into a map
+	// after the first setup we don't use any read locks
+	if circuitBreaker, ok := c.circuits[name]; ok {
+		return circuitBreaker
+	}
+	return nil
+}
+
+func (c *Manager) IsEnabled() bool {
+	return c != nil && len(c.circuits) > 0
+}
+
+type ManagerOpts struct {
+	BaseConfig              *config.CircuitBreaker
+	SubgraphCircuitBreakers map[string]*config.CircuitBreaker
+	Subgraphs               []*nodev1.Subgraph
+	FeatureFlagName         string
+	MetricStore             metric.Store
+	UseMetrics              bool
+	BaseOtelAttributes      []attribute.KeyValue
+}
+
+func NewManager(opts ManagerOpts) *Manager {
+	circuitManager := circuit.Manager{}
+
+	if opts.SubgraphCircuitBreakers == nil {
+		return &Manager{}
+	}
+
+	isBaseEnabled := opts.BaseConfig != nil && opts.BaseConfig.Enabled
+	if isBaseEnabled {
+		configuration := createConfiguration(opts.BaseConfig)
+		circuitManager.DefaultCircuitProperties = []circuit.CommandPropertiesConstructor{
+			configuration.Configure,
+		}
+	}
+
+	circuits := make(map[string]*circuit.Circuit, len(opts.Subgraphs))
+	for _, sg := range opts.Subgraphs {
+		// Base graph will start with "::"
+		sgCbName := fmt.Sprintf("%s::%s", opts.FeatureFlagName, sg.Name)
```

Comment:
Please comment from the tests regarding ff.

Reply:
Feature flags can have separate routing urls, and can fail independently (also tested this).